### ISSUE TITLE
Invalid date not detected on Chrome

### ISSUE THREAD

**Issue by lyuan-levyonline** [hyperform/hyperform#49]
An invalid date (e.g. 04/31/1998) is not caught but only in a specific situation: using `hyperform(document.forms[0])` on a Chrome browser.

If using above code on IE (the only one I've tested so far) or using `hyperform(window)` on either browser, invalid date is caught and form not submitted.

My test code:

```
<!DOCTYPE html>
<html>
<head>
	<script type="text/javascript" src="dist/hyperform.js"></script>
</head>
<body>
<form id="test">
	<p>A date field:</p>
	<input type="date">
	<input type="submit">
</form>
<script type="text/javascript">
	hyperform(document.forms[0]);
</script>
</body>
</html>
```

**Comment by Boldewyn** [comment]
Thank you for the report! I'll look into it. (And especially thanks for preparing a test code sample! :smile: )

**Comment by Boldewyn** [comment]
The good news is, that this concrete case seems to be fixed in the latest stable Chrome.

The bad news is, that all browsers, that support `type=date` natively, will hide bad input and set `input.value === ''`. Also my current approach to look at the original `validity.badInput` somehow fails (possibly because the browser doesn't like relocating native code?). If I find no way to detect the real user input, I’ll have to close the ticket without fix :disappointed:.

**Comment by Boldewyn** [comment]
There was another problem with the code, when `hyperform()` was called directly on a `form`. That's fixed now in the upcoming 0.9.21. `badInput` is detected again. (It's still not possible to access the value, that the user actually entered, though...) But at least a bad form submission will be halted now.
